.github/workflows/checks.yml

```diff
@@ -59,14 +59,23 @@ jobs:
       matrix:
         provider:
           - Docker
+          - Nutanix
           # Uncomment below once we have the ability to run e2e tests on other providers from GHA.
           # - AWS
-          # - Nutanix
+        cni:
+          - Calico
+          - Cilium
+        addon-strategy:
+          - ClusterResourceSet
+          - HelmAddon
```

Comment:
I like this splitting up! We'll need to update the required checks in canaveral to allow this PR to be merged though.

Reply:
Actually I wonder if we can just make this e2e labels instead of splitting up the provider/cni/addon-strategy like this.

We could continue to run Docker e2e as a single build in that case, and split up the longer running builds like AWS, Nutanix, etc as we need to - more flexibility. WDYT?

Reply:
That could work as well but then we'd have to extract the provider and the runs-on value for workflow conditionals from the labels in GitHub workflow steps which could get ugly.

Reply:
as discussed offline, we won't proceed with this.